xivo_dao/alchemy/tests/test_linefeatures.py

```diff
@@ -55,5 +56,50 @@ class TestApplication(DAOTestCase):
     def test_getter(self):
         application = self.add_application()
         line = self.add_line(application_uuid=application.uuid)
-
         assert_that(line.application, equal_to(application))
+
+
+class TestCallerID(DAOTestCase):
+
+    @staticmethod
+    def _caller_id_name(id):
+        return 'Test' + str(id + 1)
+
+    @staticmethod
+    def _caller_id_num(id):
+        return str(1000 + id)
+
+    def _run_query_test(self, lines, count, filter):
+        for i in range(0, count):
+            id = random.randint(0, count - 1)
+            result = (self.session.query(LineFeatures)
+                      .filter(filter(id))
+                      .first())
+            assert_that(result, equal_to(lines[id]))
+
+    def test_caller_id_sql_expression(self):
+        tests_count = 10
+        half = tests_count >> 1
+        tenant = self.add_tenant()
+        context = self.add_context(tenant_uuid=tenant.uuid)
+        line = []
+
+        sip = [
+            self.add_endpoint_sip(caller_id='"{}" <{}>'.format(self._caller_id_name(i), self._caller_id_num(i)))
+            for i in range(0, half)
+        ]
+
+        sccp = [
+            self.add_sccpline(cid_name=self._caller_id_name(i), cid_num=self._caller_id_num(i))
+            for i in range(half, tests_count)
+        ]
+
+        [line.append(self.add_line(context=context.name, endpoint_sip_uuid=sip_.uuid)) for sip_ in sip]
+        [line.append(self.add_line(context=context.name, endpoint_sccp_id=sccp_.id)) for sccp_ in sccp]
+
+        self._run_query_test(line, tests_count, lambda id: LineFeatures.caller_id_name == self._caller_id_name(id))
+        self._run_query_test(line, tests_count, lambda id: LineFeatures.caller_id_num == self._caller_id_num(id))
+        self._run_query_test(line, tests_count, lambda id: and_(
+            LineFeatures.caller_id_name == self._caller_id_name(id),
+            LineFeatures.caller_id_num == self._caller_id_num(id)
+        ))
```

Comment:
Each test should test ~one thing
Split test for each endpoint
Moreover tests should remain simple and explicit and maintainable by junior/senior. There is no difference between testing 10 lines or 2. Only a lot of loop and unnecessary complexity are added. Don't hesitate to hardcode value to improve readability (only for tests...)